### ISSUE TITLE
Inconsistent/non-standard naming convention of RxJava

### ISSUE THREAD

**Issue by infinity0** [ReactiveX/RxJava#558]
I've been taking the [Reactive Programming](https://www.coursera.org/course/reactive) course on coursera and ran into this issue. It is too late to change the method names, but perhaps you can talk about it in the documentation so newbies don't get confused. `Future` and `Promise` in Scala are similar to Guava's [ListenableFuture](http://docs.guava-libraries.googlecode.com/git-history/release/javadoc/com/google/common/util/concurrent/ListenableFuture.html) and [SettableFuture](http://docs.guava-libraries.googlecode.com/git-history/release/javadoc/com/google/common/util/concurrent/SettableFuture.html) respectively. (edit: I gather this is probably inherited from Microsoft's C# library, but I can't be bothered right this minute to register an account on codeplex)

In RxJava, the `Observer` trait has the methods {`onCompleted`, `onError`, `onNext`}. These are used to **handle an event**. Likewise, the `*Subject` classes have methods named _exactly the same_, but for a different purpose - **emitting events**.

This is inconsistent with every other use of the naming pattern `onEvent` out there (JS event handlers, even scala's `Future` itself), which typically **registers a callback** to handle the event. (An equivalent convention is `addEventListener`, used by some Java UI libraries.) By contrast, the method for firing an event is usually named `fireEvent` or `emitEvent` or just plain `Event`.

`Observer` does not register callbacks, it _is_ the callback, so its methods should be called {`handleComplete`, `handleError`, `handleNext`}. There is no direct analogy in scala (a plain Function callback takes that role in the Future ecosystem), but it is analogous to a Guava's [FutureCallback](http://docs.guava-libraries.googlecode.com/git-history/release/javadoc/com/google/common/util/concurrent/FutureCallback.html), though that suffers from the same problem.

`*Subject` is similarly inconsistent. It is an event source, that fires _callbacks registered elsewhere_, so it is analogous to a `Promise`. Its methods should be called {`complete`,`error`,`emitNext`}.

It would be good if the documentation mentioned this inconsistency in naming, perhaps even compare it with other similar libraries in a table like the one below, as a helpful guide for people to understand the concepts being discussed.

TL;DR:
- Observable, Future - registers callbacks, methods should be called `onEvent(callback)`, or `subscribe(handler)`
- Observer, FutureCallback - is the callback/handler itself, methods should be called `handleEvent` or `eventOccured`
- Subject, Promise - emits the events, methods should be called `event` or `fireEvent` or `emitEvent`
